clash-prelude/src/Clash/Magic.hs

```diff
@@ -42,3 +42,11 @@ setName
   :: forall (name :: Symbol) a . a -> name ::: a
 setName = id
 {-# NOINLINE setName #-}
+
+-- | Do not deduplicate, i.e. /keep/, an applied function inside a
+-- case-alternative; do not try to share the function between multiple
+-- branches
```

Comment:
It would be useful to add the "why" to this.

"This is useful for very small functions, where the overhead of sharing would be bigger than simply duplicating it."

Reply:
PS: How do people make these fancy commitable diff reviews? I can't find it.